### PR TITLE
feat: [BT-1750]: Update Dockerfile-ubi-java-base

### PR DESCRIPTION
**Pull request by AnshuH**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37967)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_NA_** 
 


            
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37967)

File: dockerization/base-images/Dockerfile-ubi-java-base [hunk 1]
```diff
@@ -19,5 +19,4 @@ RUN curl -OL https://github.com/adoptium/temurin11-binaries/releases/download/jd
  && /opt/harness/OpenJDK11/jdk-11.0.15+10/bin/java -version && rm -rf OpenJDK11U-jdk_x64_linux_hotspot_11.0.15_10.tar.gz
 
 ENV JAVA_HOME=/opt/harness/OpenJDK11/jdk-11.0.15+10
-ENV PATH=$JAVA_HOME/bin:$PATH
-ENV CLASSPATH_LIMIT=400000
```
Comment (USB-05) [inline dockerization/base-images/Dockerfile-ubi-java-base:23]: Is this line removed intentinally ?

**Review by USB-05 — DISMISSED** [review]

File: dockerization/base-images/Dockerfile-ubi-java-base (hunk 1 above)
Comment (USB-05) [reply dockerization/base-images/Dockerfile-ubi-java-base:23]: Since bazel version is upgraded to 5.0.0, CLASSPATH_LIMIT is not required now.

**Comment by USB-05** [comment]
Looks good to me, However with this change we are not following the best practices of Docker images and containers. Starting docker container and image in production with root user is not the correct approach.

**Review by diptiman-adak — APPROVED** [review]
> Reviewed 16 of 18 files at r2, 3 of 3 files at r4, 7 of 7 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37967)* status: Waiting on team(s): sto.

**Comment by manishjaiswal2** [comment]
We have reviewed the changes & as per discussion we should have root access with yum installer on login to pod.

**Review by manishjaiswal2 — APPROVED** [review]

**Pull request merged**